mock/py/mockbuild/buildroot.py

```diff
@@ -566,7 +566,7 @@ def nuke_rpm_db(self):
     @traceLog()
     def _open_lock(self):
         file_util.mkdirIfAbsent(self.basedir)
-        self._lock_file = open(os.path.join(self.basedir, "buildroot.lock"), "a+")
+        self._lock_file = open(os.path.join(os.path.abspath(self.basedir)+ ".lock"), "a+")
```

Comment:
This `os.path.abspath(self.basedir)` seems like a no-op. What you did is `s/buildroot.lock/.lock/`.

Reply:
Yes, I see it looks like a no-op but it is not exactly. It is probably a expensive way to get the absolute path of the basedir (irrespective of end backslashes or relative paths) and create a new file in the same parent dir of basedir:
```
self.basedir='/tmp/mock'
os.path.abspath(self.basedir)+".lock"
'/tmp/mock.lock'
```
```
self.basedir='/tmp/mock/'
os.path.abspath(self.basedir)+".lock"
'/tmp/mock.lock'
```
```
os.chdir('/tmp')
self.basedir='./mock'
os.path.abspath(self.basedir)+".lock"
'/tmp/mock.lock'
```

There are maybe more elegant solutions to it...any suggestion?

Reply:
Ah, this is the effect
```diff
- /basedir/buildroot.lock
+ /basedir.lock
```

Reply:
Exactly